shared/Microsoft.Extensions.ActivatorUtilities.Sources/ActivatorUtilities.cs

```diff
@@ -294,7 +294,7 @@ public object CreateInstance(IServiceProvider provider)
                 {
                     return _constructor.Invoke(_parameterValues);
                 }
-                catch (Exception ex)
+                catch (Exception ex) when (ex.InnerException != null)
```

Comment:
`Invoke` specifically throws `TargetInvocationException` when it was thrown by the method that's called. The other exception types indicated a mistake on the part of the code doing reflection